File: 012.jarakirefactor/java_src/Sequencia3.java
package sequencia;

import java.util.Scanner;

public class Sequencia3 {
	public static void main(String[] args){
		
		int num1, num2, soma;
		
		Scanner input = new Scanner(System.in);
		
		System.out.println("Informe o primeiro numero");
		num1 = input.nextInt();
		
		System.out.println("Informe o segundo numero");
		num2 = input.nextInt();
		
		soma = num1 + num2;
		System.out.println("A soma é: " + soma);
	}
}

File: 012.jarakirefactor/ebin/sequencia3.erl
-module(sequencia3).

-compile(export_all).

-import(loop, [for/3, while/2]).

main(V_args) ->
    st:new(),
    st:put({main, "args"}, {'String', V_args}),
    io:format("~s~n", ["Informe o primeiro numero"]),
    st:put({main, "num1"}, {int, io:fread(">", "~d")}),
    io:format("~s~n", ["Informe o segundo numero"]),
    st:put({main, "num2"}, {int, io:fread(">", "~d")}),
    st:put({main, "soma"},
	   {int, st:get(main, "num1") + st:get(main, "num2")}),
    io:format("~s~p~n",
	      ["A soma é: ", st:get(main, "soma")]).

